packages/playwright/src/runner/testServer.ts

```diff
@@ -34,33 +34,27 @@ import type { TestServerInterface, TestServerInterfaceEventEmitters } from '../i
 import { Runner } from './runner';
 import { serializeError } from '../util';
 import { prepareErrorStack } from '../reporters/base';
+import type { ConfigCLIOverrides } from '../common/ipc';
+import { loadConfig, resolveConfigFile, restartWithExperimentalTsEsm } from '../common/configLoader';
+import { webServerPluginsForConfig } from '../plugins/webServerPlugin';
+import type { TraceViewerRedirectOptions, TraceViewerServerOptions } from 'playwright-core/lib/server/trace/viewer/traceViewer';
+import type { TestRunnerPluginRegistration } from '../plugins';
 
 class TestServer {
-  private _config: FullConfigInternal;
+  private _configFile: string | undefined;
   private _dispatcher: TestServerDispatcher | undefined;
   private _originalStdoutWrite: NodeJS.WriteStream['write'];
   private _originalStderrWrite: NodeJS.WriteStream['write'];
 
-  constructor(config: FullConfigInternal) {
-    this._config = config;
+  constructor(configFile: string | undefined) {
+    this._configFile = configFile;
     process.env.PW_LIVE_TRACE_STACKS = '1';
```

Comment:
We probably do not need this one for regular "play" button in VSCode. Only need it for UI mode and/or ShowBrowser?

Reply:
you mean re-creating the config? I'd like to reuse it as much as possible though.